### ISSUE TITLE
Wiki: Basics and Attribute Mapping

### ISSUE THREAD

**Issue by SeanDbDev** [JoshClose/CsvHelper#44]
Examples are not working.  What's going on?  I'm sure I'm missing something simple here. I'm using VS2012 (VS11?) and my project is an MVC4 Internet Application.

BASICS:
var csv = new CsvReader( new StreamReader( "Actors.csv" ) );
var actorsList = csv.GetRecords<Actor>();

Design Time Error: 'CsvHelper.CsvReader' does not contain a definition for 'GetRecords' and no extension method 'GetRecords' accepting a first argument of type 'CsvHelper.CsvReader' could be found (are you missing a using directive or an assembly reference?)

I don't think so.  Am I? 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MyWebAppName.Models;
using CsvHelper;
using System.IO;

ATTRIBUTE MAPPING:
public class CustomClass
{
    [CsvField( Name = "String Field" )]
    public string StringProperty { get; set; }

```
[CsvField( Name = "Guid Field" )]
public Guid GuidProperty { get; set; }

[CsvField( Name = "Int Field" )]
public int IntProperty { get; set; }
```

}

Design Time Error: The type or namespace name 'CsvField' could not be found (are you missing a using directive or an assembly reference?)

I don't think so.  Am I? I even added a bunch to try, here is the list:
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data.Entity;
using CsvHelper;
using CsvHelper.Configuration; //Added to try
using System.ComponentModel.DataAnnotations; //Added to try
using System.ComponentModel.Design; //Added to try
using System.ComponentModel.DataAnnotations.Schema; //Added to try
using System.ComponentModel.DataAnnotations.Resources; //Added to try
using Attribute; //Added to try

HOWEVER:
I do have the following working but would much rather use the .GetRecords method:
                HttpPostedFileBase file = Request.Files[0];
                ICsvParser csvParser = new CsvParser(new StreamReader(file.InputStream));
                CsvReader csvReader = new CsvReader(csvParser);
                string[] headers = { };
                List<string[]> rows = new List<string[]>();
                string[] row;
                while (csvReader.Read())
                {
                    // Gets Headers if they exist
                    if (csvReader.FieldHeaders.Any() && !headers.Any())
                    {
                        headers = csvReader.FieldHeaders;
                    }
                    row = new string[headers.Count()];
                    for (int i = 0; i < headers.Count(); i++)
                    {
                        row[i] = csvReader.GetField(i);
                    }
                    rows.Add(row);
                }
                List<ImportItem> ImportItems = new List<ImportItem>();  
                for (int i = 0; i < rows.Count(); i++)
                {
                    ImportItem importItem = new ImportItem { 
                        ID = i + 1,
                        StringProperty= rows[i].ElementAt(Array.IndexOf(headers, "StringProperty")).ToString(),
                        DecimalProperty= Decimal.Parse(rows[i].ElementAt(Array.IndexOf(headers, "DecimalProperty"))),
                     };
                     ImportItems.Add(importItem);
                 }

Any help would be greatly appreciated.  Like I said, I just know I'm missing something silly here.

**Comment by cda-1** [comment]
@JoshClose 

I'm targeting dnx451 with an ASP.NET 5 rc1-update1 class library, and I have a similar issue to OP's cropping up. The compiler isn't able to see the "CsvClassMap" class anywhere. When I investigated with the Object Browser, I saw that it was indeed using the .net 2.0 version of the assembly. I removed and re-added the package to no avail.

However, I was able to work around the issue by going into the project.lock.json file and changing the occurrences of net20 to net40-client in all sections where it occurs, save for the last one that defines all the package files. It's a super easy fix, but it'll have to be applied every time I change my packages until I get to the bottom of it.

Any idea why this may be occurring?

**Comment by jamesbascle** [comment]
I can't sift thru github on mobile at the moment, but I recall an extremely
similar issue someone else had that cropped up for them in the past few
weeks where they had to modify the .net version in their project files as
well.

**Comment by cda-1** [comment]
@jamesbascle that comment was enough for me to dig a little further. I found the conversation in issue #449. They found they same work-around as I did, but I found an even better, dirtier hack.

Copy the "C:\Users\[your username]\.dnx\packages\CsvHelper\2.13.5\lib\net40-client" folder to
"C:\Users\[your username]\.dnx\packages\CsvHelper\2.13.5\lib\dnx451"

This works for me, possibly because my class library targets "dnx451" in my project.json framework section. Note that asp.net 5 rc1 changed it to something else recently, but I reverted it to keep EF7 migrations working in my class library.

Anyway, this hack is still dirty because you'd have to do it on every machine you develop on. However, you don't have to dig into your project.lock.json file and update it every time it is regenerated. It generates correctly with those files copied.
